lib/centoswebpanel_installer.php

```diff
@@ -0,0 +1,203 @@
+<?php
+include_once dirname(__FILE__) . DS . 'softaculous_installer.php';
+
+class CentoswebpanelInstaller extends SoftactulousInstaller
+{
+    /**
+     * Validates informations and runs a softaculous installation script on CentOS Web Panel
+     *
+     * @param stdClass $service An object representing the CentOS Web Panel service to execute a script for
+     * @param stdClass $meta The module row meta data for the service
+     * @return boolean Whether the script succeeded
+     */
+    public function install($service, $meta)
+    {
+        if (!isset($this->Clients)) {
+            Loader::loadModels($this, ['Clients']);
+        }
+        if (!isset($this->Form)) {
+            Loader::loadComponents($this, ['Form']);
+        }
+
+        // Get data for executing script
+        $serviceFields = $this->Form->collapseObjectArray($service->fields, 'value', 'key');
+        $configOptions = $this->Form->collapseObjectArray($service->options, 'value', 'option_name');
+        $client = $this->Clients->get($service->client_id);
+
+        // Login and get the cookies
+        $autoLoginData = [
+            'action' => 'list',
+            'key' => $meta->api_key,
+            'user' => $serviceFields['centoswebpanel_username'],
+            'module' => 'softaculous'
+        ];
+        $autoLoginUrl = 'https://' . $meta->host_name . ':' . $meta->port . '/v1/user_session';
```

Comment:
I notice a lot of variable properties/array keys are presumed to exist. I would think it would be more robust to check that they exist first. The function arguments can also enforce the type, in this case, stdClass for both `$service` and `$meta`.

Reply:
And if they do not exist?  There are not exactly defaults.